web/pdf_find_controller.js

```diff
@@ -742,8 +751,16 @@ class PDFFindController {
 
     // Update the match count.
     const pageMatchesCount = this._pageMatches[pageIndex].length;
-    if (pageMatchesCount > 0) {
-      this._matchesCountTotal += pageMatchesCount;
+    this._matchesCountTotal += pageMatchesCount;
+    if (this.#updateMatchesCountOnProgress) {
+      if (pageMatchesCount > 0) {
+        this.#updateUIResultsCount();
+      }
+    } else if (this.#visitedPagesCount < this._linkService.pagesCount - 1) {
+      this.#visitedPagesCount += 1;
+    } else {
```

Comment:
These two branches can be combined:
```suggestion
    } else if (++this.#visitedPagesCount === this._linkService.pagesCount) {
```

Reply:
It was to avoid the last increment... but I guess it doesn't really matter.

Reply:
I can't imagine that it matters performance-wise, and it slightly simplifies the code (for a non-default code-path).